### PR TITLE
support for arn based retrivals in secretsmanager

### PR DESCRIPTION
**Pull request by macnev2013**
## Motivation
This PR aims to add support for arn based resource retrivals for the following apis.

- `get_secret_value`
- `put_secret_value`
- `restore_secret`
- `rotate_secret`
- `delete_secret`

## Changes
- ➕ We now return data for resources that are in other account based on the ARN provided.
- ✅  added snapshot validated test case `RestoreSecret` since we are moving the responses away from moto and there were no validation tests for it.
- 🧹 Cleanup for dead code  

NOTE ⚠️ : IAM enforcement is supported in LocalStack PRO - which means we are not testing against unauthorized resource access in Community version.

## Testing
Added aws validated test cases to access secretsmanager from cross-account.

## Related PR
- https://github.com/localstack/localstack/pull/10457

### CODE REVIEW

File: tests/aws/services/secretsmanager/test_secretsmanager.py [hunk 1]
```diff
@@ -2139,3 +2161,121 @@ def test_cross_account_access(self, aws_client, secondary_aws_client, cleanups):
...
+        )
+
+        response = secondary_aws_client.secretsmanager.get_secret_value(SecretId=secret_arn)
+        assert response["SecretString"] == "new-secret"
+
+        secondary_aws_client.secretsmanager.delete_secret(
+            SecretId=secret_arn, ForceDeleteWithoutRecovery=False, RecoveryWindowInDays=7
+        )
+
+        poll_condition(
```
Comment (viren-nadkarni) [inline tests/aws/services/secretsmanager/test_secretsmanager.py:2263]: You need to add an `assert` statement here. Please update all other uses in other tests too.

File: localstack/services/secretsmanager/provider.py [hunk 2]
```diff
@@ -297,10 +369,37 @@ def update_secret(
...
+        except MotoOperationNotPermittedOnReplica:
+            raise InvalidRequestException(
+                "Operation not permitted on a replica secret. Call must be made in primary secret's region."
+            )
+        except MotoInvalidRequestException:
+            raise InvalidRequestException(
+                "An error occurred (InvalidRequestException) when calling the UpdateSecret operation: "
+                "You can't perform this operation on the secret because it was marked for deletion."
+            )
+        return UpdateSecretResponse(**json.loads(secret))
```
Comment (viren-nadkarni) [inline localstack/services/secretsmanager/provider.py:402]: Could you confirm that the responses for this (and other modified) operations are AWS validated at least once in the tests?

**Review by viren-nadkarni — CHANGES_REQUESTED** [review]

File: tests/aws/services/secretsmanager/test_secretsmanager.py [hunk 3]
```diff
@@ -179,6 +179,28 @@ def test_create_and_update_secret(self, secret_name: str, sm_snapshot, cleanups,
...
         sm_snapshot.match("delete_secret_res_1", delete_secret_res_1)
 
+    @markers.aws.validated
+    def test_secret_restore(self, secret_name: str, sm_snapshot, cleanups, aws_client):
+        cleanups.append(
+            lambda: aws_client.secretsmanager.delete_secret(
+                SecretId=secret_name, ForceDeleteWithoutRecovery=True
+            )
+        )
+        create_secret_rs = aws_client.secretsmanager.create_secret(
```
[the remark this answers is not in the document]
Comment (macnev2013) [reply tests/aws/services/secretsmanager/test_secretsmanager.py:189]: No, I haven't missed it. It's not done purposefully. We are already snapshot matching `CreateSecret` in other tests. I wanted this test to be focused on `RestoreSecret`.

File: tests/aws/services/secretsmanager/test_secretsmanager.py (hunk 1 above)
Comment (macnev2013) [reply tests/aws/services/secretsmanager/test_secretsmanager.py]: Good catch.I've updated this in the previous test as well.

File: localstack/services/secretsmanager/provider.py [hunk 4]
```diff
@@ -117,6 +124,19 @@ def get_moto_backend_for_resource(
...
+        secret_id: str, request: RequestContext, backend: SecretsManagerBackend
+    ):
+        try:
+            secret = backend.describe_secret(secret_id)
+        except MotoSecretNotFoundException:
+            raise ResourceNotFoundException("Secrets Manager can't find the specified secret.")
+        if secret.kms_key_id is None and request.account_id != secret.account_id:
+            raise InvalidRequestException(
+                "You can't access a secret from a different AWS account if you encrypt the secret with the default KMS service key."
+            )
```
[the remark this answers is not in the document]
Comment (macnev2013) [reply localstack/services/secretsmanager/provider.py:138]: Added assertions for this in AWS validated test.

**Review by viren-nadkarni — APPROVED** [review]

**Pull request merged**